solutions/kubernetes/cluster-setup.sh

```diff
@@ -22,27 +22,24 @@ source functions.sh
 
 HOST_FILE=$PWD/hosts
 SSH_KEY_FILE=/root/.ssh/id_rsa
+ALL_NODES=$(cat "$HOST_FILE" | awk -F[,] '{print $1}' | cut -d'=' -f2)
+PRIMARY_NODE=$(head -1 "$HOST_FILE" | awk -F[,] '{print $1}' | cut -d'=' -f2)
+WORKER_NODES=$(cat "$HOST_FILE" | grep -v "$PRIMARY_NODE" | awk -F[,] '{print $1}' | cut -d'=' -f2)
 UNTAINT="$1"
 
-function setup_cluster {
-
+function setup_cluster() {
     validation
     generate_rsa_key
     nodes_setup
+    k8s_deployment_type "$UNTAINT"
 
-    ALL_NODES=$(cat "$HOST_FILE" | awk -F[,] '{print $1}' | cut -d'=' -f2)
-    PRIMARY_NODE=$(head -1 "$HOST_FILE" | awk -F[,] '{print $1}' | cut -d'=' -f2)
-    WORKER_NODES=$(cat "$HOST_FILE" | grep -v "$PRIMARY_NODE" | awk -F[,] '{print $1}' | cut -d'=' -f2)
-
-    echo "---------------[ Setting up kubernetes cluster for following nodes ]--------------------------------------"
+    add_primary_separator "Setting up kubernetes cluster for following nodes"
     echo PRIMARY NODE="$PRIMARY_NODE"
     echo WORKER NODES="$WORKER_NODES"
-    echo "------------------------------------------------------------------------------------------------------"
+    add_common_separator
 
-    for node in $ALL_NODES
-	do 
-	    scp -q cluster-functions.sh functions.sh "$node":/var/tmp/
-	done
+    # Copy scripts to all nodes:
+	scp_all_nodes cluster-functions.sh functions.sh
```

Comment:
remove space before function. indent it correctly.

Reply:
Done